akka-stream/src/main/boilerplate/akka/stream/scaladsl/ZipLatestWithApply.scala.template

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (C) 2014-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream._
+import akka.stream.impl.StreamLayout
+import akka.stream.stage.{ GraphStage, GraphStageLogic, InHandler, OutHandler }
+
+trait ZipLatestWithApply {
+
+  [2..20#/**
+   * Create a new `ZipLatestWith` specialized for 1 inputs.
+   *
+   * @param zipper zipping-function from the input values to the output value
+   */
+  def apply[[#A1#], O](zipper: ([#A1#]) ⇒ O): ZipLatestWith1[[#A1#], O] =
+    new ZipLatestWith1(zipper)
+  #
+
+  ]
+
+}
+
+[2..20#/** `ZipLatestWith` specialized for 1 inputs */
+class ZipLatestWith1[[#A1#], O] (val zipper: ([#A1#]) ⇒ O) extends GraphStage[FanInShape1[[#A1#], O]] {
+  override def initialAttributes = Attributes.name("ZipLatestWith1")
+  override val shape: FanInShape1[[#A1#], O] = new FanInShape1[[#A1#], O]("ZipLatestWith1")
+  def out: Outlet[O] = shape.out
+  [#val in0: Inlet[A1] = shape.in0#
+  ]
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+      // Without this field the completion signalling would take one extra pull
+      var willShutDown = false
+
+      [#var hasValue0 = false#
+      ]
+      [#var value0 : A1 = _#
+      ]
+      var hasLast  = false
+      [#var last0: A1 = _#
+      ]
+      var waitingForTuple = false
+
+      setHandler(
+        out,
+        new OutHandler {
+          override def onPull(): Unit = {
+            if (hasAllValues) {
+              pushIfFirstTupleOrDifferentTuple()
+            } else {
+              waitingForTuple = true
+            }
+            pullAllIfNeeded()
+          }
+        }
+      )
+
+      [#setHandler(
+        in0,
+        new InHandler {
+          override def onPush() = {
+            value0 = grab(in0)
+            hasValue0 = true
+            if (waitingForTuple && hasAllValues) {
+                pushTuple()
+                waitingForTuple = false
+                pullAllIfNeeded()
+            }
+          }
+
+          override def onUpstreamFinish(): Unit = {
+            if (!isAvailable(in0)) completeStage()
+            willShutDown = true
+          }
+        }
+      )#
+      ]
+
+      private def hasAllValues = [#hasValue0#&&]
+
+      private def pushTuple(): Unit = {
+        push(out, zipper([#value0#,]))
+        if (willShutDown) completeStage()
+        [#last0 = value0#
+        ]
+        hasLast = true
+      }
+
+      private def pushIfFirstTupleOrDifferentTuple() =
+        if (hasLast) {
+          if ([#value0 != last0#||]) pushTuple() else waitingForTuple = true
+        } else {
+          pushTuple()
+        }
```

Comment:
Actually this check is to avoid re-emitting another, identical tuple directly upon pull. The corresponding test is: `does not emit the same pair upon two pulls`

The scenario you describe should be covered, see the corresponding `emit even if pair is the same` test. 

Let me know if you want me to add more to these tests.

Reply:
I think it'd need to be the same instance of pair, not an equal pair then?

Reply:
yes, adding